### PR TITLE
Add Vmotion Guidelines to EKS-A docs

### PR DESCRIPTION
**Pull request by mimcdevitt**
*Issue #, if available:*
[2572](https://github.com/aws/eks-anywhere-internal/issues/2572)

*Description of changes:*
Added section to vsphere troubleshooting to give guidance on using VMware vmotion with EKS-A clusters.

*Testing (if applicable):*

*Documentation added/planned (if applicable):*
Adding section to vsphere troubleshooting.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 1]
```diff
@@ -988,6 +988,101 @@ $ kubectl get deployments -n eksa-system
 NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
 eksa-controller-manager   1/1     1            1           4h13m
 ```
+## VMware vSphere Best Practices for vMotion with EKS Anywhere (EKS-A)
+
+#### Introduction
+
+EKS Anywhere (EKS-A) enables organizations to deploy and manage Kubernetes clusters on-premises, including in VMware vSphere environments. VMware vMotion is a feature within vSpher
```
Comment (abhay-krishna) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:995]:
```suggestion
EKS Anywhere (EKS-A) enables organizations to deploy and manage Kubernetes clusters on-premises, including in VMware vSphere environments. VMware vMotion is a feature within vSphere
```

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 2]
```diff
@@ -988,6 +988,101 @@ $ kubectl get deployments -n eksa-system
...
+e that allows live migration of virtual machines (VMs) between ESXi hypervisor hosts. This document outlines the guidelines  for using vMotion to migrate EKS-A nodes between vSpher
```
Comment (abhay-krishna) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:996]:
```suggestion
that allows live migration of virtual machines (VMs) between ESXi hypervisor hosts. This document outlines the guidelines  for using vMotion to migrate EKS-A nodes between vSpher
```

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 3]
```diff
@@ -988,6 +988,101 @@ $ kubectl get deployments -n eksa-system
...
+e ESXi hosts using vMotion while ensuring cluster stability.
+
+#### Considerations for EKS-A Node Migration using vMotion
+
+When migrating EKS-A nodes with vMotion, several considerations must be kept in mind, particularly around configuration values defined in the [EKS-A vSphere cluster spec file](http
+s://anywhere.eks.amazonaws.com/docs/getting-started/vsphere/vsphere-spec/). These configurations must remain unchanged during the migration and the infrastructure these configurati
```
Comment (abhay-krishna) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:1002]:
```suggestion
s://anywhere.eks.amazonaws.com/docs/getting-started/vsphere/vsphere-spec/). These configurations must remain unchanged during the migration and the infrastructure these configurations
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Shared storage is a requirement for vmotion of EKS-A clusters.  Storage such as vSAN, Fiber Channel SAN, or NFS should be shared between the supporting vSphere ESXi hosts for maintaining access to the VM's backing data without relying on storage vMotion, which is not supported in EKS-A environments.
+
+
+3. **Monitoring Before and After Migration**
+
+To verify cluster health and node stability,  monitor the EKS-A nodes and pods before and after the vMotion migration:
+
+* Before migration, run the following commands to check the current health and status of the EKS-A nodes and pods.
+
+```
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:74]:
```suggestion
  ```
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    kubectl get nodes
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:75]:
```suggestion
  kubectl get nodes
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    kubectl get pods -a
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:76]:
```suggestion
  kubectl get pods -a
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+* After vMotion activity is completed, run the command agains to verify that the nodes and pods are still operational and healthy.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:79]:
```suggestion
  * After vMotion activity is completed, run the command again to verify that the nodes and pods are still operational and healthy.
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+
+4. **Infrastructure Maintenance During vMotion**
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:82]:
```suggestion
* **Infrastructure maintenance during vMotion**
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+It is recommended that no other infrastructure maintenance activities be performed during the vMotion operation. The underlying datacenter infrastructure supporting the network, storage, and server resources utilized by VMware vSphere must remain stable during the vMotion process. Any interruptions in these services could lead to partial or complete failures in the vMotion process, potentially causing the EKS-A nodes to lose connectivity or experience disruptions in normal operations.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:84]:
```suggestion
  It is recommended that no other infrastructure maintenance activities be performed during the vMotion operation. The underlying datacenter infrastructure supporting the network, storage, and server resources utilized by VMware vSphere must remain stable during the vMotion process. Any interruptions in these services could lead to partial or complete failures in the vMotion process, potentially causing the EKS Anywhere nodes to lose connectivity or experience disruptions in normal operations.
```

**Review by chrisnegus — COMMENTED** [review]
Really good work. Most of my comments are formatting and style issues. I like the new Nodes section. As you suggested, I think it will be a good place to put more content for managing EKS Anywhere nodes.

File: docs/content/en/docs/clustermgmt/nodes/_index.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
+---
+title: "Nodes"
+linkTitle: "Nodes"
+date: 2017-01-05
+weight: 30
+description: >
+  Managing EKS-A nodes
```
[the remark this answers is not in the document]
Comment (mimcdevitt) [reply docs/content/en/docs/clustermgmt/nodes/_index.md:7]: Thank you very much,  will avoid using EKS-A in the future.

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+description: >
+  Using vMotion with  EKS Anywhere clusters
+aliases:
+---
+
+## vMotion with EKS Anywhere
+
+#### Introduction
+
+EKS Anywhere (EKS-A) enables organizations to deploy and manage Kubernetes clusters on-premises, including in VMware vSphere environments. VMware vMotion is a feature within vSphere that allows live migration of virtual machines (VMs) between ESXi hypervisor hosts. This document outlines the guidelines for using vMotion to migrate EKS-A nodes between vSphere ESXi hosts using vMotion while ensuring cluster stability.
```
[the remark this answers is not in the document]
Comment (mimcdevitt) [reply docs/content/en/docs/clustermgmt/nodes/vmotion.md:13]: Agree, not needed.

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+#### Considerations for EKS-A Node Migration using vMotion
```
[the remark this answers is not in the document]
Comment (mimcdevitt) [reply docs/content/en/docs/clustermgmt/nodes/vmotion.md:15]: Thank you

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+When migrating EKS-A nodes with vMotion, several considerations must be kept in mind, particularly around configuration values defined in the [EKS-A vSphere cluster spec file](https://anywhere.eks.amazonaws.com/docs/getting-started/vsphere/vsphere-spec/). These configurations must remain unchanged during the migration and the infrastructure these configurations represent should also not change.
```
[the remark this answers is not in the document]
Comment (mimcdevitt) [reply docs/content/en/docs/clustermgmt/nodes/vmotion.md:17]: Thank you very much for all the assistance

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+
+1. **No Cross-vCenter vMotion**
```
[the remark this answers is not in the document]
Comment (mimcdevitt) [reply docs/content/en/docs/clustermgmt/nodes/vmotion.md:20]: Thanks,  I agree. I had them as bulleted at one point not sure why I changed them

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+EKS-A nodes cannot be migrated between different vCenter environments using vMotion. The nodes must remain within the same vCenter instance for proper EKS-A operation. The vCenter Server managing the EKS-A cluster is specified in the `VSphereDatacenterConfig` section of the EKS-A cluster spec file, under the `spec.server` field and cannot be changed.
```
[the remark this answers is not in the document]
Comment (mimcdevitt) [reply docs/content/en/docs/clustermgmt/nodes/vmotion.md:22]: Thanks for fixing the missing relref

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+
+4. **Node Network Configuration Stability**
+
+The IP address, subnet mask, and default gateway of each EKS-A node must remain unchanged during the vMotion process. Any modifications to the IP address configuration can cause communication failures between the EKS-A nodes, pods, and the control plane, leading to disruptions in EKS-A cluster operations.
+
+
+5. **EKS-A Configuration Stabiltiy**
+
+EKS-A environment itself should remain unchanged during vmotion.  Do not perform or trigger any EKS-A changes or life cycle events while performing vmotion.
```
[the remark this answers is not in the document]
Comment (mimcdevitt) [reply docs/content/en/docs/clustermgmt/nodes/vmotion.md:48]: should be vMotion, I will fix

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+### Considerations for node migration using vMotion
+
+When migrating EKS Anywhere nodes with vMotion, several considerations must be kept in mind, particularly around configuration values defined in the [vSphere cluster spec file]({{< relref "/docs/getting-started/vsphere/vsphere-spec" >}}) . These configurations must remain unchanged during the migration and the infrastructure these configurations represent should also not change.
+
+
+* **No cross-vCenter vMotion**
+
+EKS Anywhere nodes cannot be migrated between different vCenter environments using vMotion. The nodes must remain within the same vCenter instance for proper EKS Anywhere operation. The vCenter Server managing the EKS Anywhere cluster is specified in the `VSphereDatacenterConfig` section of the EKS Anywhere [vSphere cluster spec file]({{< relref "/docs/getting-started/vsphere/vsphere-spec" >}}), under the `spec.server` field, and cannot be changed.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:22]:
```suggestion
  EKS Anywhere nodes cannot be migrated between different vCenter environments using vMotion. The nodes must remain within the same vCenter instance for proper EKS Anywhere operation. The vCenter Server managing the EKS Anywhere cluster is specified in the `VSphereDatacenterConfig` section of the EKS Anywhere [vSphere cluster spec file]({{< relref "/docs/getting-started/vsphere/vsphere-spec" >}}), under the `spec.server` field, and cannot be changed.
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+
+### Best practices for vMotion with EKS Anywhere clusters
+
+* **Follow VMware vMotion best practices**
+
+  * General best practices: Review VMware's general guidelines for optimal vMotion performance, such as ensuring sufficient CPU, memory, and network resources, and minimizing load on the ESXi hosts during the migration. Refer to the [VMware vMotion documentation](https://docs.vmware.com/) for details.
+
+  * VMware vMotion Networking Best Practices: Whenever possible, follow the [Networking Best Practices for VMware vMotion](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vcenterhost.doc/GUID-7DAD15D4-7F41-4913-9F16-567289E22977.html) to optimize performance and reduce the risk of issues during the migration process.
+
+  * Use High-Speed Networks: A 10GbE or higher speed network is recommended to ensure smooth vMotion operations for EKS-A nodes, particularly those with large memory footprints.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:59]:
```suggestion
  * Use High-Speed Networks: A 10GbE or higher speed network is recommended to ensure smooth vMotion operations for EKS Anywhere nodes, particularly those with large memory footprints.
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+
+
+* **Shared Storage**
+
+  Shared storage is a requirement for vmotion of EKS-A clusters.  Storage such as vSAN, Fiber Channel SAN, or NFS should be shared between the supporting vSphere ESXi hosts for maintaining access to the VM's backing data without relying on storage vMotion, which is not supported in EKS-A environments.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:64]:
```suggestion
  Shared storage is a requirement for vmotion of EKS Anywhere clusters.  Storage such as vSAN, Fiber Channel SAN, or NFS should be shared between the supporting vSphere ESXi hosts for maintaining access to the VM's backing data without relying on storage vMotion, which is not supported in EKS Anywhere environments.
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+
+
+* **Monitoring before and after migration**
+
+  To verify cluster health and node stability,  monitor the EKS-A nodes and pods before and after the vMotion migration:
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:69]:
```suggestion
  To verify cluster health and node stability,  monitor the EKS Anywhere nodes and pods before and after the vMotion migration:
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+  * Before migration, run the following commands to check the current health and status of the EKS-A nodes and pods.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:71]:
```suggestion
  * Before migration, run the following commands to check the current health and status of the EKS Anywhere nodes and pods.
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+  * After vMotion activity is completed, run the commands again to verify that the nodes and pods are still operational and healthy.
+
+```
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:74]:
```suggestion
  ```
```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+  kubectl get nodes
+  kubectl get pods -a
+```
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:77]:
```suggestion
  ```
```

**Review by chrisnegus — COMMENTED** [review]
Just a couple more EKS-A and indent corrections.

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+
+
+* **Monitoring before and after migration**
+
+  To verify cluster health and node stability,  monitor the EKS Anywhere nodes and pods before and after the vMotion migration:
+
+  * Before migration, run the following commands to check the current health and status of the EKS Anywhere nodes and pods.
+  * After vMotion activity is completed, run the commands again to verify that the nodes and pods are still operational and healthy.
+
+  kubectl get nodes
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:74]: Sorry, something weird happened here. I tried to do two spaces in front of the three backticks before and after the two kubectl lines. Somehow they got deleted. Can you add them back in around those two lines and indent them two spaces?

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+  Shared storage is a requirement for vmotion of EKS Anywhere clusters.  Storage such as vSAN, Fiber Channel SAN, or NFS should be shared between the supporting vSphere ESXi hosts for maintaining access to the VM's backing data without relying on storage vMotion, which is not supported in EKS Anywhere environments.
+
+
+* **Monitoring before and after migration**
+
+  To verify cluster health and node stability,  monitor the EKS Anywhere nodes and pods before and after the vMotion migration:
+
+  * Before migration, run the following commands to check the current health and status of the EKS Anywhere nodes and pods.
+  * After vMotion activity is completed, run the commands again to verify that the nodes and pods are still operational and healthy.
+
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/nodes/vmotion.md:73]:
```suggestion
  ```

```

File: docs/content/en/docs/clustermgmt/nodes/vmotion.md (hunk 24 above)
Comment (mimcdevitt) [reply docs/content/en/docs/clustermgmt/nodes/vmotion.md:76]: restored backticks

**Pull request merged**